src/stateless_validation.cpp

```diff
@@ -1,48 +1,140 @@
+#include <algorithm>
+#include <string>
+#include <vector>
+
 #include "veriblock/arith_uint256.hpp"
 #include "veriblock/consts.hpp"
 #include "veriblock/stateless_validation.hpp"
+#include "veriblock/strutil.hpp"
 
 namespace VeriBlock {
 
 static const ArithUint256 MAXIMUM_DIFFICULTY(
     "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF");
 
-bool checkProofOfWork(const BtcBlock& block) {
+bool checkBitcoinTransactionForPoPData(const VbkPopTx& tx,
+                                       ValidationState& state) {
+  WriteStream stream;
+  tx.publishedBlock.toRaw(stream);
+  tx.address.getPopBytes(stream);
```

Comment:
why `getPopBytes` and not `toRaw`?

Reply:
getPopBytes() returns a subset of a data from the toRaw(), some special 'PopBytes'